src/composables/UserAuthentication.js

```diff
@@ -1,90 +1,55 @@
-
-    userManager.signinSilent()
-      .catch(() => {
-        userManager.signinRedirect();
-      });
-  });
+export function getUserSessionToken() {
+  return localStorage.getItem('sessionToken');
 }
 
 /**
- * Initiate the login process by redirecting the user to the authentication provider.
- * @returns {Promise<void>} Promise with nothing on success otherwise an error.
+ * Log in the user using its temporaryCode. Once the user is logged, it saves
+ * the user information in the dedicated store.
+ * @param {string} temporaryCode - The temporary code obtained from the github provider.
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
  */
-export async function login() {
-  return userManager.signinRedirect();
-}
+export async function login(temporaryCode) {
+  return api.login(temporaryCode)
+    .then((response) => {
+      const userStore = useUserStore();
 
-/**
- * Complete the login process by handling the callback after the user
- * is redirected back from the authentication provider.
- * @returns {Promise<object>} Promise with the logged-in user on success otherwise an error.
- */
-export async function completeLogin() {
-  return userManager.signinRedirectCallback();
-}
+      setUserSessionToken(response.data.sessionToken);
 
-/**
- * Complete the silent login process by handling the callback after the user
- * is refreshed in the background without any user interaction.
- * @returns {Promise<object>} Promise with the refreshed user information on success,
- * otherwise an error.
- */
-export async function completeSilentLogin() {
-  return userManager.signinSilentCallback();
+      userStore.username = response.data.username;
+      userStore.firstname = response.data.firstname;
+    })
+    .catch(() => { throw new Error('Login Failed'); });
 }
 
 /**
- * Retrieve the current user if available or triggers a request
- * to fetch the user's information from the authentication provider.
- * @returns {Promise<object>} Promise with the current user on success otherwise an error.
+ * Retrieve (GET request) information about the current user and store them in the dedicated store.
+ * @param {string} sessionToken - The current user's session token.
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
  */
-export async function getUser() {
-  return userManager?.getUser().then((user) => (user
-    ? new User({
-      firstname: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.firstname} }}`, user),
-      lastname: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.lastname} }}`, user),
-      email: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.email} }}`, user),
-      id: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.id} }}`, user),
+export async function getUserInfo(sessionToken) {
```

Comment:
you don't return userInformation at the end of .then

Reply:
so change name by initUserInformations or return userInformation. But I prefer first solution.